gradle-baseline-java/src/main/java/com/palantir/baseline/plugins/BaselineModuleJvmArgs.java

```diff
@@ -365,70 +395,110 @@ public abstract static class ModuleJvmArgsArgumentProvider implements CommandLin
+
+            List<String> args = Stream.concat(exportsArgs, opensArgs).toList();
 
             log.debug(
-                    "BaselineModuleJvmArgs executing {} on project {} with exports: {}",
-                    getTaskName().getOrNull(),
-                    getProjectPath().getOrNull(),
+                    "BaselineModuleJvmArgs configuring {} with exports: {}",
+                    getDescription().get(),
                     args);
-            return args;
-        }
 
-        private static ImmutableList<String> compilationArgs(Stream<String> allExports, Stream<String> allOpens) {
-            return Stream.concat(allExports, allOpens)
-                    .distinct()
-                    .sorted()
-                    .flatMap(ModuleJvmArgsArgumentProvider::addExportArg)
-                    .collect(ImmutableList.toImmutableList());
+            return args;
         }
 
-        private static ImmutableList<String> runtimeArgs(Stream<String> allExports, Stream<String> allOpens) {
-            Stream<String> exportsArgs =
-                    allExports.distinct().sorted().flatMap(ModuleJvmArgsArgumentProvider::addExportArg);
-            Stream<String> opensArgs = allOpens.distinct().sorted().flatMap(ModuleJvmArgsArgumentProvider::addOpensArg);
-            return Stream.concat(exportsArgs, opensArgs).collect(ImmutableList.toImmutableList());
+        public final Set<String> allModulesPackagePairUnnamed() {
+            return Sets.union(getExports().get(), getOpens().get()).stream()
+                    .map(ModuleJvmArgsArgumentProvider::appendAllUnnamed)
+                    .collect(Collectors.toSet());
         }
 
         private static Stream<String> addExportArg(String modulePackagePair) {
-            return Stream.of("--add-exports", modulePackagePair + "=ALL-UNNAMED");
+            return Stream.of("--add-exports", appendAllUnnamed(modulePackagePair));
         }
 
         private static Stream<String> addOpensArg(String modulePackagePair) {
-            return Stream.of("--add-opens", modulePackagePair + "=ALL-UNNAMED");
+            return Stream.of("--add-opens", appendAllUnnamed(modulePackagePair));
+        }
+
+        private static String appendAllUnnamed(String modulePackagePair) {
+            return modulePackagePair + "=ALL-UNNAMED";
+        }
+
+        public static ModuleJvmArgsArgumentProvider fromJustClasspath(
+                Task task, Callable<FileCollection> classpathCallable) {
+            return create(task).configureWithClasspath(classpathCallable);
+        }
+
+        public static ModuleJvmArgsArgumentProvider fromJustExtensionForCompilation(Task task) {
+            ModuleJvmArgsArgumentProvider argumentProvider = create(task);
+
+            argumentProvider.getExports().addAll(extension(task).exports());
+            // For compilation, `--add-opens` does nothing at all. Instead, each of the `opens` needs to
+            // become an export for compilation to work.
```

Comment:
this doesn't make sense? it sounds like an improperly configured extension i.e. you haven't properly specified your exports

Reply:
okay, I think I understand now. `--add-opens` when used at runtime, implies `--add-export` so you don't need to also specify `--add-export` at runtime. So if you used directly imported an encapsulated method (requires `--add-exports`) and also did deep reflection on it (requires `--add-opens`) you'd only need to specify `--add-opens`. 

This kinda sucks imo, but it is what it is.

Reply:
Can we update the comment however